contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,199 @@
+The framework aims to make development and testing of enterprise web applications easier. Angular is maintained by Google, as well as a community of developers.
+
+>Angular started as a side project. Back in 2009, Miško Hevery and Adam Abrons released a project that would help developers, build web applications using simple HTML tags. The name “Angular” comes from the angle brackets, or < >, that surround all HTML tags.
+>
+>Miško described the idea behind the framework in an <a href="https://www.infoworld.com/article/2612801/javascript/whats-so-special-about-googles-angularjs.html">interview done in 2013</a>. 
+
+
+## Why learn Angular?
+
+Angular is a complex framework and it can be challenging for new developers to understand.
+But the advantages Angular provides justifies the steep learning curve.
+
+1. **Angular supports Single Page Applications (SPAs).**
+
+    <box>Single Page Applications are a type of web applications that load a single HTML page, and the page is updated dynamically according to the interaction of the user. SPAs can communicate with the back-end servers without refreshing the full web page, for loading data in the application. They provide a better user experience.</box>
+The fact that Angular supports the development of SPAs, is a good reason to learn the framework.
+ 
+2. **Angular uses TypeScript.** Angular applications are built using TypeScript language, a superset of JavaScript. TypeScript provides static typing. This helps the compiler show warnings about any potential errors in the code, before the app is even run. Another advantage of TypeScript is that it provides code completion using IntelliSense. IntelliSense provides active hints as code is added.
+
+    <box> TypeScript was created and is maintained by Microsoft, and has been <a href="https://developer.telerik.com/featured/the-rise-of-typescript/">growing in popularity</a> for the last few years; therefore you can feel at ease when using it in your application.</box>
+                                                  
+3. **Google Long-Term Support.** Some software engineers consider the fact that Angular is supported by Google an advantage of the technology. While this may sound justified, Google itself is not enough. The good sign though is that Google announced Long-Term Support (LTS) for the technology back in 2017.
+
+    [source](https://hackr.io/blog/why-should-you-learn-angular-in-2018)
+
+### Benefit: Angular provides a declarative user interface.
+
+Angular uses HTML to define the app’s user interface. HTML is a declarative language which is more intuitive than using imperative code to define the UI. Imperative code requires you to define <i>how to perform an action</i> that is describe the flow. Declarative code, on the other hand enables you to describe the logic without worrying about the implementation. How does it help? You don’t need to invest your time in program flows and in deciding what loads first.
```

Comment:
> Imperative code requires you to define <i>how to perform an action</i> that is describe the flow.

Phrasing needs to be corrected.

> You don’t need to invest your time in program flows and in deciding what loads first.

You do not need to invest time in dealing with how your program runs and deciding what to load first.